ffi/input_android.lua

```diff
@@ -198,6 +198,14 @@ function input.waitForEvent(sec, usec)
                         android.lib.AInputQueue_finishEvent(android.app.inputQueue, event[0], handled)
                     end
                 end
+            elseif poll_state == C.LOOPER_ID_USER then
+                local message = ffi.new("unsigned char [4]")
+                C.read(fd[0], message, 4)
+                if message[0] == C.AEVENT_POWER_CONNECTED then
+                    commandHandler(C.AEVENT_POWER_CONNECTED, 0)
+                elseif message[0] == C.AEVENT_POWER_DISCONNECTED then
+                    commandHandler(C.AEVENT_POWER_DISCONNECTED, 0)
```

Comment:
Note for future enhancements, in case the first-byte-is-a-command design evolves: reading an int is perfectly valid.

e.g.,

```
local message = ffi.new("uint32_t")
C.read(fd[0], message, ffi.sizeof(message))
```